src/core/reference/include/openvino/reference/reduce_mean.hpp

```diff
@@ -26,6 +26,10 @@ void reduce_mean(const T* in, T* out, const Shape& in_shape, const AxisSet& redu
     reduce_sum(in, out, in_shape, reduction_axes);
 
     const auto out_shape = util::reduce(in_shape, reduction_axes);
+    if (shape_size(in_shape) == 0) {
+        return;
+    }
+
     const auto out_size = shape_size(out_shape);
     const auto count = static_cast<T>(shape_size(in_shape) / out_size);
     std::transform(out, std::next(out, out_size), out, [count](const T value) {
```

Comment:
```suggestion
    const auto out_size = shape_size(out_shape);
    const auto count = static_cast<T>(shape_size(in_shape) / out_size);
    if (count > 1){
        std::transform(out, std::next(out, out_size), out, [count](const T value) {
```
Count can be used to avoid div by 0

Reply:
But if `out_size` is zero, it will still try to divide by zero when calculating the `count`.